src/main/java/ru/practicum/shareit/request/service/RequestServiceImplementation.java

```diff
@@ -0,0 +1,88 @@
+package ru.practicum.shareit.request.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import ru.practicum.shareit.exception.ObjectNotFoundException;
+import ru.practicum.shareit.item.dto.ItemForRequestDto;
+import ru.practicum.shareit.item.dto.mapper.ItemMapper;
+import ru.practicum.shareit.item.model.Item;
+import ru.practicum.shareit.item.storage.ItemRepository;
+import ru.practicum.shareit.request.dto.NewRequestDto;
+import ru.practicum.shareit.request.dto.RequestDto;
+import ru.practicum.shareit.request.dto.RequestGetAllDto;
+import ru.practicum.shareit.request.dto.mapper.RequestMapper;
+import ru.practicum.shareit.request.model.Request;
+import ru.practicum.shareit.request.storage.RequestRepository;
+import ru.practicum.shareit.user.model.User;
+import ru.practicum.shareit.user.storage.UserRepository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+public class RequestServiceImplementation implements RequestService {
+    private final RequestRepository requestRepository;
+    private final UserRepository userRepository;
+    private final RequestMapper requestMapper;
+    private final ItemRepository itemRepository;
+    private final ItemMapper itemMapper;
+
+    @Override
+    public RequestDto addNewRequest(NewRequestDto newRequestDto, int userId) {
+        User user = checkIfUserExists(userId);
+        Request request = requestMapper.mapNewRequestDtoToRequest(newRequestDto);
+        request.setRequestor(user);
+        request.setCreated(LocalDateTime.now());
+        return requestMapper.mapRequestToRequestDto(requestRepository.save(request));
+    }
+
+    @Override
+    public List<RequestGetAllDto> getOwnRequests(int userId) {
+        checkIfUserExists(userId);
+        List<Request> requests = requestRepository.findByRequestorIdOrderByCreatedDesc(userId);
+        List<Item> items = itemRepository.findAllByRequestIdIn(requests.stream()
+                .map(Request::getId).collect(Collectors.toList()));
+        return requests.stream()
+                .map(request -> mapToRequestGetAllDto(request, items)).collect(Collectors.toList());
```

Comment:
```suggestion
                .map(request -> mapToRequestGetAllDto(request, items))
                .collect(Collectors.toList());
```
Поправь, пожалуйста, и в методах ниже

Reply:
Исправлено.